agent/src/main/scala/za/co/absa/atum/agent/AtumAgent.scala

```diff
@@ -47,25 +47,41 @@ class AtumAgent private() {
    *  @return
    */
   def getOrCreateAtumContext(atumPartitions: AtumPartitions): AtumContext = {
-    contexts.getOrElse(atumPartitions, new AtumContext(atumPartitions, this))
+    getContextOrElse(
```

Comment:
I would maybe extract the `AtumContext.fromDTO()` call into a val so that it's not that nested